common/lib/xmodule/xmodule/js/src/video/035_video_accessible_menu.js

```diff
@@ -1,241 +1,73 @@
-                case KEY.UP:
-                    this.previousMenuItemLink(this.menuItemsLinks, index).focus();
-                    break;
-                    // Scroll down  menu, wrapping at the bottom. Keep menu
-                    // open.
-                case KEY.DOWN:
-                    this.nextMenuItemLink(this.menuItemsLinks, index).focus();
-                    break;
-                    // Close menu.
-                case KEY.TAB:
-                    this.closeMenu();
-                        // TODO
-                        // What has to happen here? In speed menu, tabbing backward
-                        // will give focus to Play/Pause button and tabbing
-                        // forward to Volume button.
-                    break;
-                    // Close menu, give focus to button and change
-                    // file type.
-                case KEY.ENTER:
-                case KEY.SPACE:
-                    this.button.focus();
-                    this.changeFileType.call(this, event);
-                    this.closeMenu();
-                    break;
-                    // Close menu and give focus to speed control.
-                case KEY.ESCAPE:
-                    this.closeMenu();
-                    this.button.focus();
-                    break;
-                }
-                return false;
-            }
-            else {
-                switch (keyCode) {
-                    // Open menu and focus on last element of list above it.
-                case KEY.ENTER:
-                case KEY.SPACE:
-                case KEY.UP:
-                    this.openMenu();
-                    this.menuItemsLinks.last().focus();
-                    break;
-                    // Close menu.
-                case KEY.ESCAPE:
-                    this.closeMenu();
-                    break;
-                }
-                // We do not stop propagation and default behavior on a TAB
-                // keypress.
-                return event.keyCode === KEY.TAB;
-            }
-                                                },
-
-        setValue: function(value) {
-            this.value = value;
-            this.menuItems
-                .removeClass('active')
-                .find("a[data-value='" + value + "']")
-                .parent()
-                .addClass('active');
-        },
-
-        changeFileType: function(event) {
             var fileType = $(event.currentTarget).data('value'),
-                data = {'transcript_download_format': fileType};
+                data = {transcript_download_format: fileType},
+                url = $(event.currentTarget).attr('href'),
+                that = this;
 
-            this.setValue(fileType);
-            this.options.storage.setItem('transcript_download_format', fileType);
+            event.preventDefault();
 
             $.ajax({
-                url: this.options.saveStateUrl,
+                async: false,
```

Comment:
I don't think there is any reason to set async to false. You are using callbacks, and the risk of locking up the UI seems too high.

http://api.jquery.com/jquery.ajax/
> async (default: true)
> Type: Boolean
> By default, all requests are sent asynchronously (i.e. this is set to true by default). If you need synchronous requests, set this option to false. Cross-domain requests and dataType: "jsonp" requests do not support synchronous operation. Note that synchronous requests may temporarily lock the browser, disabling any actions while the request is active. As of jQuery 1.8, the use of async: false with jqXHR ($.Deferred) is deprecated; you must use the success/error/complete callback options instead of the corresponding methods of the jqXHR object such as jqXHR.done().